machine/v1/types_nutanixprovider.go

```diff
@@ -0,0 +1,244 @@
+
+	// Status is the status of the condition.
+	Status corev1.ConditionStatus `json:"status"`
+
+	// LastProbeTime is the last time we probed the condition.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty"`
+
+	// LastTransitionTime is the last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+
+	// Reason is a unique, one-word, CamelCase reason for the condition's last transition.
+	// +optional
+	Reason NutanixMachineProviderConditionReason `json:"reason,omitempty"`
+
+	// Message is a human-readable message indicating details about last transition.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+// RawExtensionFromNutanixMachineProviderSpec marshals the machine provider spec.
+func RawExtensionFromNutanixMachineProviderSpec(spec *NutanixMachineProviderConfig) (*runtime.RawExtension, error) {
+	if spec == nil {
+		return &runtime.RawExtension{}, nil
+	}
+
+	var rawBytes []byte
+	var err error
+	if rawBytes, err = json.Marshal(spec); err != nil {
+		return nil, fmt.Errorf("error marshalling providerSpec: %v", err)
+	}
+
+	return &runtime.RawExtension{
+		Raw: rawBytes,
+	}, nil
+}
+
+// RawExtensionFromNutanixMachineProviderStatus marshals the machine provider status
+func RawExtensionFromNutanixMachineProviderStatus(status *NutanixMachineProviderStatus) (*runtime.RawExtension, error) {
+	if status == nil {
+		return &runtime.RawExtension{}, nil
+	}
+
+	var rawBytes []byte
+	var err error
+	if rawBytes, err = json.Marshal(status); err != nil {
+		return nil, fmt.Errorf("error marshalling providerStatus: %v", err)
+	}
+
+	return &runtime.RawExtension{
+		Raw: rawBytes,
+	}, nil
+}
+
+// NutanixMachineProviderSpecFromRawExtension unmarshals a raw extension into an NutanixMachineProviderConfig type
+func NutanixMachineProviderSpecFromRawExtension(rawExtension *runtime.RawExtension) (*NutanixMachineProviderConfig, error) {
+	if rawExtension == nil {
+		return &NutanixMachineProviderConfig{}, nil
+	}
+
+	spec := new(NutanixMachineProviderConfig)
+	if err := json.Unmarshal(rawExtension.Raw, &spec); err != nil {
+		return nil, fmt.Errorf("error unmarshalling providerSpec: %v", err)
+	}
+
+	klog.V(5).Infof("Got provider Spec from raw extension: %+v", spec)
+	return spec, nil
+}
+
+// NutanixMachineProviderStatusFromRawExtension unmarshals a raw extension into an NutanixMachineProviderStatus type
+func NutanixMachineProviderStatusFromRawExtension(rawExtension *runtime.RawExtension) (*NutanixMachineProviderStatus, error) {
+	if rawExtension == nil {
+		return &NutanixMachineProviderStatus{}, nil
+	}
+
+	providerStatus := new(NutanixMachineProviderStatus)
+	if err := json.Unmarshal(rawExtension.Raw, providerStatus); err != nil {
+		return nil, fmt.Errorf("error unmarshalling providerStatus: %v", err)
+	}
+
+	klog.V(5).Infof("Got provider Status from raw extension: %+v", providerStatus)
+	return providerStatus, nil
+}
```

Comment:
Code like this is not allowed in the API repo, this must be moved into your machine-api-provider repository

Reply:
Okay, will move these util functions to the mapi-nutanix repo.

Reply:
Moved to mapi-nutanix repo.